src/thunderbots/software/geom/util.h

```diff
@@ -338,6 +338,41 @@ std::vector<Point> lineIntersection(const Segment &a, const Segment &b);
  */
 Point reflect(const Point &v, const Point &n);
 
+/**
+ * Calculates the intersection point (if it exists) of an moving element with a line
+ * defined by 2 Points
+ *
+ * @param element_velocity: X/Y velocity of the element, defined in meters per second
+ * @param element_position: X/Y position of the element, defined in meters
+ * @param line_start: X/Y point where the line begins, defined in meters
+ * @param line_end: X/Y point where the line ends, defined in meters
+ * @param boundry_area: Defines the rectangular geometry that limits the scope of
```

Comment:
Would be good to include more detail about what limiting the scope means here. Does the point of intersection need to be within the rectangle? Outside the rectangle? Does the `element_position` need to be inside? outside? etc.

Reply:
Good point, I elaborated on the scope.